definitions/checks/foreman_installer/installer_noop_run.rb

```diff
@@ -0,0 +1,22 @@
+module Checks
+  module ForemanInstaller
+    class InstallerNoopRun < ForemanMaintain::Check
+      metadata do
+        description 'Run the installer in noop mode for system checks'
+        tags :pre_upgrade
+      end
+
+      def run
+        check_error_codes = [102, 103]
+        cmd_status, output = feature(:installer).run_with_status(' -n')
+        msg = "\nThe system does not comply with checks in the installer!\n"\
+              "Fix the requirements:\n#{output}"
+        if check_error_codes.include?(cmd_status)
+          fail! msg
+        elsif cmd_status == 1
+          fail! output
```

Comment:
Why only on code 1? all non-zero exit codes should be treated as a problem.

Looking at https://github.com/theforeman/kafo#exit-code I wonder if we should have reused code 20 instead of 102 and 103. Besides kafo there are also exit codes in `foreman-installer` but various are not `1` (see `grep fail_and_exit hooks`).

Reply:
oh yes, sorry for the miss

Reply:
Added 20 error code and rest all are getting caught in elseif block.